### PR TITLE
Add option to decide panic or not when key exceed bound

### PR DESCRIPTION
**Pull request by zhangjinpeng1987**
Signed-off-by: zhangjinpeng1987 <zhangjinpeng@pingcap.com>

## What have you changed? (mandatory)

Previously TiKV will panic when encounter key exceed bound error, add a configuration to control panic or return error in this condition.

## What are the type of the changes? (mandatory)

The currently defined types are listed below, please pick one of the types for this PR by removing the others:
- Improvement (change which is an improvement to an existing feature)

## How has this PR been tested? (mandatory)

Unit test

## Does this PR affect documentation (docs) update? (mandatory)

No

## Does this PR affect tidb-ansible update? (mandatory)

No

### CODE REVIEW

**Comment by zhangjinpeng1987** [comment]
@breeswish I have tried, and it will pass the config to destination through a very long path, it is more ugly than current way.

File: src/config.rs [hunk 1]
```diff
@@ -1240,7 +1242,11 @@ impl TiKvConfig {
     }
 
     pub fn write_to_file<P: AsRef<Path>>(&self, path: P) -> Result<(), IoError> {
-        let content = toml::to_string(&self).unwrap();
+        let content = match toml::to_string(&self) {
+            Ok(content) => content,
+            Err(e) => panic!("toml to string error: {:?}", e),
+        };
+        //let content = toml::to_string(&self).unwrap();
```
Comment (nolouch) [inline src/config.rs]: remove this line.

Comment (zhangjinpeng1987) [reply src/config.rs]: Done

File: src/config.rs [hunk 2]
```diff
@@ -1044,6 +1045,7 @@ impl Default for TiKvConfig {
             log_level: slog::Level::Info,
             log_file: "".to_owned(),
             log_rotation_timespan: ReadableDuration::hours(24),
+            panic_when_key_exceed_bound: false,
```
Comment (breeswish) [inline src/config.rs]: Should the default value be true?

File: src/util/mod.rs [hunk 3]
```diff
@@ -45,6 +45,16 @@ pub mod timer;
 pub mod transport;
 pub mod worker;
 
+static PANIC_WHEN_KEY_EXCEED_BOUND: AtomicBool = AtomicBool::new(false);
+
+pub fn panic_when_key_exceed_bound() -> bool {
+    PANIC_WHEN_KEY_EXCEED_BOUND.load(Ordering::SeqCst)
```
Comment (siddontang) [inline src/util/mod.rs]: I think we can use Relaxed here.

Comment (siddontang) [reply src/util/mod.rs]: following PANIC_MARK can use Relaxed too

File: src/config.rs (hunk 2 above)
Comment (zhangjinpeng1987) [reply src/config.rs]: No, we will set it true in our internal testings, but in user environment we should alert not panic.

File: src/util/mod.rs (hunk 3 above)
Comment (zhangjinpeng1987) [reply src/util/mod.rs]: I think it is ok, because it is very very rarely happen.

**Pull request merged**